### PR TITLE
Allows the AI to use the autodoc

### PR DESCRIPTION
**Pull request by DominikPanic**
Also had to do a slight interface update cleanup since such a tiny change made me find quite a few issues.

Removed updateDialog since it was redundant.
Mass replaced all occurences with updateUsrDialog
Neither of those can nor should be used in process due to the way interface interaction works, and hacking it together is not really feasible. Interfaces that want to update constantly should be using nanoUI instead.

### CODE REVIEW

File: code/game/objects/machinery/autodoc.dm [hunk 1]
```diff
@@ -1041,176 +1038,171 @@
...
-				if(!needed)
-					N.fields["autodoc_manual"] += create_autodoc_surgery(null,LIMB_SURGERY,ADSURGERY_INTERNAL,1)
-				updateUsrDialog()
-
-			if(href_list["broken"])
-				for(var/datum/limb/L in connected.occupant.limbs)
-					if(L)
-						if(L.limb_status & LIMB_BROKEN)
-							N.fields["autodoc_manual"] += create_autodoc_surgery(L,LIMB_SURGERY,ADSURGERY_BROKEN)
+	if(connected.occupant && ishuman(connected.occupant))
```
Comment (Rohesie) [inline code/game/objects/machinery/autodoc.dm]:
```suggestion
	if(ishuman(connected.occupant))
```

File: code/game/objects/machinery/autodoc.dm [hunk 2]
```diff
@@ -1041,176 +1038,171 @@
...
+		// manual surgery handling
+		var/datum/data/record/N = null
+		for(var/datum/data/record/R in GLOB.datacore.medical)
```
Comment (Rohesie) [inline code/game/objects/machinery/autodoc.dm]: No need for typechecking.

File: code/game/objects/machinery/autodoc.dm [hunk 3]
```diff
@@ -1041,176 +1038,171 @@
...
+			N.fields["autodoc_manual"] += create_autodoc_surgery(null,EXTERNAL_SURGERY,ADSURGERY_BLOOD)
+
+		if(href_list["organgerms"])
+			N.fields["autodoc_manual"] += create_autodoc_surgery(null,ORGAN_SURGERY,ADSURGERY_GERMS)
+
+		if(href_list["eyes"])
+			N.fields["autodoc_manual"] += create_autodoc_surgery(null,ORGAN_SURGERY,ADSURGERY_EYES,0,connected.occupant.internal_organs_by_name["eyes"])
+
+		if(href_list["organdamage"])
+			for(var/datum/limb/L in connected.occupant.limbs)
```
Comment (Rohesie) [inline code/game/objects/machinery/autodoc.dm]: No need for typechecking.

File: code/game/objects/machinery/autodoc.dm [hunk 4]
```diff
@@ -1041,176 +1038,171 @@
...
+				if(L)
```
Comment (Rohesie) [inline code/game/objects/machinery/autodoc.dm]: Check unecessary inside the `for` loop.

File: code/game/objects/machinery/autodoc.dm [hunk 5]
```diff
@@ -1041,176 +1038,171 @@
...
-				for(var/datum/limb/L in connected.occupant.limbs)
-					if(L)
-						if(L.limb_status & LIMB_NECROTIZED)
-							N.fields["autodoc_manual"] += create_autodoc_surgery(L,LIMB_SURGERY,ADSURGERY_NECRO)
+			if(!needed)
+				N.fields["autodoc_manual"] += create_autodoc_surgery(null,ORGAN_SURGERY,ADSURGERY_DAMAGE,1)
+
+		if(href_list["internal"])
+			for(var/datum/limb/L in connected.occupant.limbs)
+				if(L)
```
Comment (Rohesie) [inline code/game/objects/machinery/autodoc.dm]: Check and typecheck unnecessary,

File: code/game/objects/machinery/autodoc.dm (hunk 4 above)
Comment (DominikPanic) [reply code/game/objects/machinery/autodoc.dm]: Not my code

File: code/game/objects/machinery/autodoc.dm (hunk 5 above)
Comment (DominikPanic) [reply code/game/objects/machinery/autodoc.dm]: Same, just reduced indentation. But I'll fix it regardless.

**Pull request merged**